packages/curvenote/src/web/deploy.ts

```diff
@@ -151,23 +158,129 @@ export async function deployContentToCdn(session: ISession, opts?: { ci?: boolea
   session.log.info(toc(`☁️  Uploaded ${files.length} files in %s.`));
 
   const cdnKey = uploadTargets.id;
+  return { cdnKey };
+}
 
+/**
+ * Upload content to the (private) staging bucket
+ *
+ * @param session
+ * @param opts
+ * @returns cdnkey and filepaths for deployment
+ */
+async function uploadContent(session: ISession, opts?: { ci?: boolean }) {
+  const { files, uploadRequest } = await prepareUploadRequest(session);
+  const { json: uploadTargets } = await session.post<SiteUploadResponse>('/sites/upload', {
+    ...uploadRequest,
+  });
+  const { cdnKey } = await processUpload(session, files, uploadTargets, opts);
+  return { cdnKey, filepaths: files.map(({ to }) => ({ path: to })) };
+}
+
+/**
+ * Deploy content to the public/private CDN based on the usePublicCdn argument
+ *
+ * @param session
+ * @param usePublicCdn whether to deploy to the public CDN or default private CDN
+ * @param cdnKey
+ * @param filepaths
+ * @returns cdnkey
```

Comment:
This `cdnKey` is just the same as the `cdnKey` input. Doesn't change when content is deployed.